test/AlgebraicGeometry/Schemes/smoothness_test.jl

```diff
@@ -0,0 +1,47 @@
+@testset "_is_smooth" begin
+  R, (x, y, z) = QQ[:x, :y, :z]
+
+  I1 = ideal(R, x*y)
+  IA3 = spec(R)
+  X1, _ = sub(IA3, I1)
+  @test !Oscar._is_smooth(X1)
+  @test !Oscar._is_smooth(X1; jacobian_cut_off=0)
```

Comment:
@afkafkafk13 : I think I have now implemented a truly generic smoothness test here which can also handle non-reduced, non-equidimensional, and non-connected cases. I'm still a bit skeptic, since all the classical implementations seem to solve this via some primary/equidimensional decomposition which I seem to be able to fully avoid. If you have the time: Could you have a look and tell me what you think? Maybe you have some further exotic example in mind for which my implementation fails. If so, it would be good to know about it.